tests/test_sshfsmount.cpp

```diff
@@ -188,167 +219,126 @@ struct SshfsMount : public mp::test::SftpServerTest
     REPLACE(ssh_channel_request_exec, request_exec);
 
     EXPECT_THROW(make_sshfsmount(), std::runtime_error);
-    EXPECT_TRUE(invoked);
+    EXPECT_TRUE(invoked_fail);
 }
 
-TEST_F(SshfsMount, throws_when_uid_is_not_an_integer)
+TEST_P(SshfsMountExecute, test_succesful_invocation)
 {
-    bool invoked{false};
-    auto request_exec = [&invoked](ssh_channel, const char* raw_cmd) {
-        std::string cmd{raw_cmd};
-        if (cmd.find("id -u") != std::string::npos)
-        {
-            invoked = true;
-        }
-        return SSH_OK;
-    };
-    REPLACE(ssh_channel_request_exec, request_exec);
+    std::string target = GetParam().first;
+    CommandVector commands = GetParam().second;
 
-    std::string output{"ubuntu"};
-    auto remaining = output.size();
-    auto channel_read = make_channel_read_return(output, remaining, invoked);
-    REPLACE(ssh_channel_read_timeout, channel_read);
-
-    EXPECT_THROW(make_sshfsmount(), std::invalid_argument);
-    EXPECT_TRUE(invoked);
+    test_command_execution(commands, target);
 }
 
-TEST_F(SshfsMount, throws_when_unable_to_obtain_gid)
+TEST_P(SshfsMountExecuteThrow, test_invalid_arg_when_executing)
 {
-    bool uid_invoked{false};
-    bool gid_invoked{false};
-    auto request_exec = [this, &uid_invoked, &gid_invoked](ssh_channel, const char* raw_cmd) {
-        std::string cmd{raw_cmd};
-        if (cmd.find("id -u") != std::string::npos)
-        {
-            uid_invoked = true;
-        }
-        else if (cmd.find("id -g") != std::string::npos)
-        {
-            uid_invoked = false;
-            gid_invoked = true;
-            exit_status_mock.return_exit_code(SSH_ERROR);
-        }
-        return SSH_OK;
-    };
-    REPLACE(ssh_channel_request_exec, request_exec);
-
-    std::string output{"1000"};
-    auto remaining = output.size();
-    auto channel_read = make_channel_read_return(output, remaining, uid_invoked);
-    REPLACE(ssh_channel_read_timeout, channel_read);
-
-    EXPECT_THROW(make_sshfsmount(), std::runtime_error);
-    EXPECT_TRUE(gid_invoked);
+    MP_ASSERT_THROW_THAT(test_command_execution(GetParam().first), std::exception,
+                         Property(&std::exception::what, ContainsRegex(GetParam().second)));
 }
 
-TEST_F(SshfsMount, throws_when_gid_is_not_an_integer)
+//
+// Instantiate the parameterized tests suites from above.
+//
+
+INSTANTIATE_TEST_SUITE_P(SshfsMountThrowWhenError, SshfsMountFail,
+                         testing::Values("mkdir", "chown", "id -u", "id -g", "cd", "pwd"));
+
+// Commands to check that a version of FUSE smaller that 3 gives a correct answer.
+CommandVector fuse_cmds = {{"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "FUSE library version: 2.9.0\n"},
+                           {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -o slave -o transform_symlinks -o "
+                            "allow_other -o nonempty :\"source\" \"target\"",
+                            "don't care\n"}};
```

Comment:
We don't have an explicit test for when the version is 3 or higher, it is just tested via `default_cmds`. Please think whether we should add a version of the above with version: 3.0.0 and higher?

Reply:
Good catch. Added the test.